src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service;
+
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private Map<String, Integer> storage = new HashMap<>();
```

Comment:
The storage field is public, which violates the encapsulation principle. Make the storage field private and provide a getter for it if necessary. This is a code quality issue.